agent-launcher/src/com/thoughtworks/go/agent/launcher/AgentLauncherImpl.java

```diff
@@ -57,6 +58,7 @@ public AgentLauncherImpl(AgentProcessParentRunner agentProcessParentCreator) {
 
     public int launch(AgentLaunchDescriptor descriptor) {
         Thread shutdownHook = null;
+        new SystemEnvironment().initializeAgentLogDir();
```

Comment:
Don't think this is required. It won't help people who don't upgrade their bootstrappers, If someone does use a new bootstrapper, then, the location will either be logs or /var/log/go-agent anyway. Which is why I didn't modify the`agent-bootstrapper-log4j.properties`to use the system property. But, if you think we must do it just for the sake of consistency...